app/src/sharedTest/java/org/oppia/android/app/onboarding/CreateProfileFragmentTest.kt

```diff
@@ -0,0 +1,433 @@
+          closeSoftKeyboard()
+        )
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      // No screen change as the navigation to the next screen is not implemented yet.
+      // This should fail in the future once the screen has been implemented.
+      onView(withId(R.id.create_profile_nickname_label))
+        .check(
+          matches(
+            withText(
+              context.getString(
+                R.string.create_profile_activity_nickname_label
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @Config(qualifiers = "land")
+  @Test
+  fun testFragment_landscapeMode_continueButtonClicked_emptyNickname_showNicknameErrorText() {
+    launchNewLearnerProfileActivity().use {
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withText(R.string.create_profile_activity_nickname_error))
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Config(qualifiers = "land")
+  @Test
+  fun testFragment_landscape_continueButtonClicked_afterErrorShown_launchesLearnerIntroScreen() {
+    launchNewLearnerProfileActivity().use {
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withText(R.string.create_profile_activity_nickname_error))
+        .check(matches(isDisplayed()))
+
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .perform(
+          editTextInputAction.appendText("John"),
+          closeSoftKeyboard()
+        )
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      // No screen change as the navigation to the next screen is not implemented yet.
+      // This should fail in the future once the screen has been implemented.
+      onView(withId(R.id.create_profile_nickname_label))
+        .check(
+          matches(
+            withText(
+              context.getString(
+                R.string.create_profile_activity_nickname_label
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_tapToAddPictureClicked_hasGalleryIntent() {
+    launchNewLearnerProfileActivity().use {
+      onView(withText(R.string.create_profile_activity_profile_picture_prompt))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+      intended(hasAction(Intent.ACTION_PICK))
+    }
+  }
+
+  @Config(qualifiers = "land")
+  @Test
+  fun testProfileProgressFragment_imageSelectAvatar_configChange_checkGalleryIntent() {
```

Comment:
Is there a way to test that an image was successfully picked & processed by the fragment presenter, as well?

Reply:
I have added `testFragment_tapToAddPictureClicked_loadsTheImageFromGallery` to verify that the image was picked, but this does not cover the presenter processing it because The presenter  functionality is not public in the test.

Reply:
Could we verify that an image load request was sent to the image loader? I think there are other tests in the codebase that verify this sort of flow using [TestGlideImageLoader](https://github.com/oppia/oppia-android/blob/develop/utility/src/main/java/org/oppia/android/util/parser/image/TestGlideImageLoader.kt).

Reply:
Thanks, I found a way to verify this.